selfdrive/camerad/cameras/camera_common.cc

```diff
@@ -345,58 +345,46 @@ std::thread start_process_thread(MultiCameraState *cameras, const char *tname,
 void common_camera_process_front(SubMaster *sm, PubMaster *pm, CameraState *c, int cnt) {
   const CameraBuf *b = &c->buf;
 
-  static int meteringbox_xmin = 0, meteringbox_xmax = 0;
-  static int meteringbox_ymin = 0, meteringbox_ymax = 0;
+  static int x_min = 0, x_max, y_min, y_max = 0;
   static const bool rhd_front = Params().read_db_bool("IsRHD");
 
-  sm->update(0);
-  if (sm->updated("driverState")) {
-    auto state = (*sm)["driverState"].getDriverState();
-    float face_prob = state.getFaceProb();
-    float face_position[2];
-    face_position[0] = state.getFacePosition()[0];
-    face_position[1] = state.getFacePosition()[1];
-
-    // set front camera metering target
-    if (face_prob > 0.4) {
-      int x_offset = rhd_front ? 0:b->rgb_width - 0.5 * b->rgb_height;
-      meteringbox_xmin = x_offset + (face_position[0] + 0.5) * (0.5 * b->rgb_height) - 72;
-      meteringbox_xmax = x_offset + (face_position[0] + 0.5) * (0.5 * b->rgb_height) + 72;
-      meteringbox_ymin = (face_position[1] + 0.5) * (b->rgb_height) - 72;
-      meteringbox_ymax = (face_position[1] + 0.5) * (b->rgb_height) + 72;
-    } else { // use default setting if no face
-      meteringbox_ymin = b->rgb_height * 1 / 3;
-      meteringbox_ymax = b->rgb_height * 1;
-      meteringbox_xmin = rhd_front ? 0:b->rgb_width * 3 / 5;
-      meteringbox_xmax = rhd_front ? b->rgb_width * 2 / 5:b->rgb_width;
-    }
-  }
-
   // auto exposure
   if (cnt % 3 == 0) {
-    // use driver face crop for AE
-    int x_start, x_end, y_start, y_end;
-    int skip = 1;
+    if (sm->update(0) > 0) {
```

Comment:
explicitly check if driverState was updated, otherwise this breaks if more services are added